proto/ibc/lightclients/wasm/v1/tx.proto

```diff
@@ -27,3 +30,18 @@ message MsgStoreCodeResponse {
   // the sha256 hash of the stored code
   bytes checksum = 1;
 }
+
+// MsgRemoveCodeHash defines the request type for the MsgRemoveCodeHash rpc.
+message MsgRemoveCodeHash {
+  option (cosmos.msg.v1.signer) = "signer";
+
+  string signer = 1;
+  // code hash to be removed from the store
+  bytes code_hash = 2;
+}
+
+// MsgStoreCodeResponse defines the response type for the StoreCode rpc
+message MsgRemoveCodeHashResponse {
+  // true if the code hash was found in the store
+  bool found = 1;
```

Comment:
What would be the benefit of signalling here if the code hash was not found instead of erroring in the RPC handler?

Reply:
I think the goal of remove is to ensure the item is not on the list. I don't think it constitutes an error if the item was not in there, or reversion tx.

Reply:
I see your point, but I could also argue that for the `MigrateContract` RPC if the code hash is the same instead of error, we could also just return with a successful response, since the outcome of the operation is the same as what the user asked for, which was to migrate to that other code hash (that turns out to be the same one that it already had).

Reply:
yup, I'd agree with Carlos from POV it should be similar to what we do for migrate.

Reply:
I think part of the outcome of the `MigrateContract` is the calling of the migrate entry point in the new code. Since returning early wouldn't do that, I wanted to return an error. I could see a case for it if we called the migrate entry point in the new code.

Reply:
Carlo's comment is w.r.t returning nil or err in migrate contract when new code hash is equal to old. In both cases we return early, unless your point just flew over my head.

Reply:
Ok. If one more person agrees I'll change it XD. It is worth noting that in most APIs I've seen, `Remove` functions don't return an error if the key is not present, including keyset's remove api, and the KVStore's. But I think this might make sense in a cosmos tx I suppose.

Reply:
I'd suggest removing the bool from the response. It seems unnecessary, if it's requested by users then we can add it later with no overhead.

Reply:
Done. I have removed found, and now returning an error instead if not found